godtossing/Pos/Ssc/GodTossing/Toss/Base.hs

```diff
@@ -472,10 +481,27 @@ checkCertificatesPayload
 checkCertificatesPayload epoch certs = do
     richmenSet <- getKeys <$> (note (NoRichmen epoch) =<< getRichmen epoch)
     exceptGuardM CertificateAlreadySent
-        (notM hasCertificateToss) (HM.keys certs)
+        (notM hasCertificateToss) (HM.keys (getVssCertificatesMap certs))
     exceptGuardSnd CertificateNotRichmen
         ((`HS.member` richmenSet) . addressHash . vcSigningKey)
-        (HM.toList certs)
+        (HM.toList (getVssCertificatesMap certs))
+    existingVssKeys <-
+        HS.fromList . map vcVssKey . toList . getVssCertificatesMap <$>
```

Comment:
`getVssCertificatesMap` is probably useless, because `toList` is defined for `VssCertificatesMap` too (AFAIK), but it's not really essential.